prow/gcs/reporter/reporter_test.go

```diff
@@ -255,40 +255,28 @@ func (ta *testAuthor) NewWriter(ctx context.Context, bucket, path string, overwr
 	return &testAuthorWriteCloser{author: ta}
 }
 
-func TestReportJobState(t *testing.T) {
+func TestReportJobFinished(t *testing.T) {
 	completionTime := &metav1.Time{Time: time.Date(2010, 10, 10, 19, 00, 0, 0, time.UTC)}
 	tests := []struct {
 		jobState       prowv1.ProwJobState
-		expectedFile   string
 		completionTime *metav1.Time
 		passed         bool
 	}{
-		{
-			jobState:     prowv1.TriggeredState,
-			expectedFile: "started.json",
-		},
-		{
-			jobState: prowv1.PendingState,
```

Comment:
We should keep these test cases to ensure we don't report `finished.json` before the job completes.

Reply:
The function being tested here doesn't actually check check that (it used to, but it was removed for being redundant). It's handled by `reportJobState`, which is now annoying to unit test because I have to rejig my harness to support varying numbers of files being written. Ah well...

Reply:
As it happens I have added the check back anyway because apparently if you try reporting an incomplete job it panics.

Reply:
Yeah we should have the function that reads the completion time return a bool or error to indicate if the job is complete instead of having a separate `Complete()` function. That would make it harder to forget to check `Complete()` before reading the completion time.
/shrug